README.md

```diff
@@ -1,5 +1,164 @@
-# FLEDGE Polyfill
+# FLEDGE Shim
 
 We're building a pure-JavaScript implementation of the [FLEDGE
-spec](https://github.com/WICG/turtledove/blob/master/FLEDGE.md), built on top of
-existing browser APIs.
+spec](https://github.com/WICG/turtledove/blob/master/FLEDGE.md), on top of
+existing browser APIs. The goal is to allow testing as much of FLEDGE as
+possible, in as realistic a manner as possible, given the constraint of not
+being able to add new features to the browser itself.
+
+## Status
+
+This project is just beginning.  Don't expect anything to work yet!
+
+## Design
+
+FLEDGE requires a way to store information in the browser that is (a) accessible
+across all websites but (b) only through JavaScript access control.
+`localStorage` in a cross-origin iframe fits this well. In Chrome this is not
+partitioned and only JavaScript running within the iframe can read or modify the
+data.
+
+The shim is divided into two pieces:
+
+* A *binary* that runs in a cross origin iframe, such as
+  `https://fledge-shim.example/v/1234`.
+
+* A *library* that consumers compile into their code, and which communicates
+  with the binary over `postMessage`.
+
+Almost all of the work happens in the binary; the library is a small adapter
+that translate the API from functions to messages.
+
+## API
+
+We're planning to implement the API as closely as possible to what is presented
+in the spec, but some aspects necessarily differ due to the constraints of
+running on publisher and advertiser pages, and implementing in pure JS.
```

Comment:
By pure JS do you mean post or pre compile? I'm planning on adding a webpack config in the first PR, so we wouldn't be writing it in pure js

Reply:
clarified